framework/include/controls/ControllableItem.h

```diff
@@ -0,0 +1,176 @@
+ * In general, an ControllableItem will have a one-to-one relationship with an input parameter
+ * value, but in some instances it is desirable to connect parameters with different names together.
+ * For example, within MOOSE when a material is defined in an input file multiple Material objects
+ * are generated automatically. If a parameters is controlled on one of these objects it is
+ * necessary to also have the values on the other controlled as well. This example is the
+ * driver behind the creation of this intermediate class.
+ */
+class ControllableItem
+{
+public:
+  ControllableItem(const MooseObjectParameterName & name, libMesh::Parameters::Value * value);
+  ~ControllableItem() = default;
+
+  ControllableItem(const ControllableItem &) = default;
+  ControllableItem(ControllableItem &&) = default;
+
+  ControllableItem & operator=(const ControllableItem &) = delete;
+  ControllableItem & operator=(ControllableItem &&) = delete;
+
+  /**
+   * Connects the supplied item with this item to allow for multiple parameters to be changed by
+   * one.
+   */
+  void connect(ControllableItem * item, bool type_check = true);
+
+  /*
+   * Set the value(s) of the controlled parameters stored in this class.
+   *
+   * The 'skip_type_check' flag allows this object to work with ControllableParameter that
+   * can store values of varying types.
+   */
+  template <typename T>
+  void set(const T & value, bool type_check = true);
+
+  /**
+   * Return a copy of all values for this "item".
+   */
+  template <typename T>
+  std::vector<T> get(bool type_check = true) const;
+
+  ///@{
+  /**
+   * Use the master name for comparison operators to allow object to work within a set/map.
+   */
+  bool operator==(const ControllableItem & rhs) const { return name() == rhs.name(); }
+  bool operator!=(const ControllableItem & rhs) const { return name() != rhs.name(); }
+  bool operator<(const ControllableItem & rhs) const { return name() < rhs.name(); }
+  ///@}
+
+  /**
+   * Return the name of the master parameter.
+   */
+  virtual const MooseObjectParameterName & name() const;
+
+  /**
+   * Return the type of the master parameter.
+   */
+  std::string type() const;
+
+  /**
+   * Returns a string displaying the parameter name and current value.
+   */
+  virtual std::string dump(unsigned int indent = 0) const;
+
+  ///@{
+  /**
+   * Methods for ControlOutput::outputChangedControls, these don't have meaning outside of this
+   * function.
+   */
+  void resetChanged(){ _changed = false; }
+  bool isChanged(){ return _changed; }
+  ///@}
+
+  /// Allows this to be used with std:: cout
+  friend std::ostream & operator<<(std::ostream & stream, const ControllableItem & obj);
+
+protected:
+
+  /**
+   * Constructor for creating an empty item (see ControllableAlias)
+   */
+  ControllableItem();
+
+  /// List of names for this item
+  std::vector<std::pair<MooseObjectParameterName, libMesh::Parameters::Value *>> _pairs;
+
+  /// Flag for ControlOutput
```

Comment:
This does not really help. If the comment was not here, I would be able to get the same information just by reading the code. May be you want to say, for example, under what circumstances is this flag changed, or so.